src/azul/chalice.py

```diff
@@ -467,6 +495,36 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
+    def threshold(self, *, errors: int, throttles: int):
+        def wrapper(f):
+            f.errors_threshold = errors
+            f.throttles_threshold = throttles
+            return f
+
+        return wrapper
+
+    def metric_thresholds(self) -> list[MetricThreshold]:
+        default_threshold = 0
+        thresholds = []
+        lambda_name, _ = config.unqualified_resource_name(self.app_name)
+        for metric in LambdaMetric:
+            # The api_handler lambda functions (indexer & service) aren't included
+            # in the app_module's handler_map, so we account for those first.
+            thresholds.append(MetricThreshold(lambda_name=lambda_name,
+                                              metric=metric,
+                                              value=default_threshold))
+            for handler in self.handler_map.values():
+                if isinstance(handler, chalice.app.EventSourceHandler):
+                    threshold = getattr(handler, f'{metric.name}_threshold', default_threshold)
+                    # PyCharm doesn't know `name` is an attribute of the handler object
+                    # since we dynamically add it to the object in _register_handler()
+                    # hence why we use getattr() here instead of `handler.name`
```

Comment:
```suggestion
                    # We added the `name` attribute dynamically in _register_handler() so to avoid a PyCharm type warning we use `getattr` here.
```

a little less verbose